---
title: "SC_RNA_treated vs untreated"
author: "Sneha Challa"
date: "3/9/2022"
output:
  pdf_document: default
  html_document: default
---

__* Objective: *__  

The goal of this project is to determine the heterogeniety of the tumor cells coming from the liver tissue of patients suffering with Hepatoblastoma(HB), a form of pediatric liver cancer. A secondary objective is to account for any batch effects that might be observed due to the integration of samples from multiple sources. 

__*Data: *__  

The data is obtained from NCBI geo, accession number: GSE180665. The data is downloaded and unzipped to show data from three types of tissues: actual tumor, background (i.e., cells from around the tumor) and patient derived xenografts(PDX) which are cells grown within a mouse to study different stages of cancer and its progression. 

The data is from three individuals labelled HB17, HB30 and HB53. The dataset contains both background and PDX data for patients HB30 and HB53. HB17 has data from all three tissues including tumor samples that is missing for the other two patients. Because data is from multiple sources (patients and tissues), a larger variation is expected to arise from the batch affects which can confound the actual variation in gene expression data. This report describes how we accounted for these batch effects using the conventional seruat batch processing method. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

The libraries included for the analysis of this dataset are shown below: 
```{r libraries, include=TRUE, echo=FALSE, warning=FALSE}
#Load Libraries:
library(Seurat)
library(ggplot2)
library(tidyverse)
library(gridExtra)
```

*Step 1:*  The first step is to __read the data__ into R using the `list.dirs` function. This is useful as there are 7 directories in total for each of the sample from each patient. A seurat object is created for each of the directories, generating a total of 7 seurat objects each named with the number of the patient and the kind of cells. For example one of the seurat object is named: `HB17_PDX`.
```{r data, include=FALSE, echo=FALSE, warning=FALSE}
#get data location into R: 

dirs <- list.dirs(path = '/Users/snehachalla/Desktop/Projects/Genomics/SCRNA', recursive = F, full.names = F)

dirs

#go into each folder and fetch the counts matrix, create a Seurat object and assign to a variable

for (x in dirs[-1]){
  #create variable names:
  name <- gsub('_filtered_feature_bc_matrix', '',x)
  cts <- ReadMtx(mtx = paste0('/Users/snehachalla/Desktop/Projects/Genomics/SCRNA/',x,'/matrix.mtx.gz'),
          features = paste0('/Users/snehachalla/Desktop/Projects/Genomics/SCRNA/',x,'/features.tsv.gz'),
          cells = paste0('/Users/snehachalla/Desktop/Projects/Genomics/SCRNA/',x, '/barcodes.tsv.gz'))
          
  #Create a Seurat Object and assigning a value to name:
   assign(name, CreateSeuratObject(counts = cts))
}
```

*Step 2:* The second step involves __combining these 7 seurat objects__ into a single merged object called `merged_seurat` to reduce errors and to allow for easy accession and manipulation. 
```{r merge, include=FALSE, echo=FALSE, warning=FALSE}
#Merge all the seven different Seurat objects in to one to enable easy QC: 

 merged_seurat <- merge(HB17_background, y = c(HB17_PDX, HB17_tumor, HB30_PDX, HB30_tumor,HB53_background, HB53_tumor), add.cell.ids = ls()[3:9], project = "HB")

#merged_seurat
```

The merged seurat object contains a total of 33538 features or genes across 77936 samples from the seven individual seurat objects and can be viewed by calling the object 'merged_seurat`. To access the metadata within the merged object, we use : `merged_seurat@meta.data`. The metadata at this point contains only three columns: cell identity, ncount_RNA (which is the number of molecules within a cell, including and not limited to carbohydrates, proteins, lipids, nucleic acids etc..) and nfeature_RNA (which is the number of genes). 

```{r QC, include=FALSE, echo=FALSE, warning=FALSE}

#View(merged_seurat@meta.data)
merged_seurat$sample <- rownames(merged_seurat@meta.data)
#View(merged_seurat@meta.data)

#Split sample columns

merged_seurat@meta.data <- separate(merged_seurat@meta.data, col = 'sample', into = c('Patient', "Cell_type", "Barcode"), sep = "_")

```

We can further customize the metadata table by splitting the identity column into patient type, cell type and barcode using the `separate` command in R. To check if the data has merged correctly, we can use the following sanity checks: 
```{r}
#unique(merged_seurat@meta.data$Patient)
#unique(merged_seurat@meta.data$Cell_type)
```
 
*Step 3:* The next step involves __Data Exploration__. There are multiple steps/stages of data exploration that can be performed to get a better understanding of the data. The data can then be filtered based on a number of criteria such as mitochondrial gene percentages, ribosomal gene percentanges, doublet percentage, transcript count threshold, number of highly and lowly expressed genes etc. To demonstrate a few, here we see how to:

*a) Calculate the percentage of mitochondrial genes:*
```{r mito, include=FALSE, echo=FALSE, warning=FALSE}
merged_seurat$percent.mt <- PercentageFeatureSet(merged_seurat, pattern = '^MT-')

#View(merged_seurat@meta.data)
```
This gives us a list of mitochondrial genes for each of the 77936 samples and adds the column called `percent.mt` to the metadata. The more number of mitochondrial genes  We can also visualize this using a violin plot. The violin plot below shows that a major proportion of cells have multiple molecules in the range of about 30,000 per cell. A very few cells have even higher number of molecules, approximately around 60,000 to 90,000 per cell. Similarly, most of the cells have a large number of unique genes about 7000 genes per cell. A very small proportion of cells have even greater number of genes i.e. around 10,000 to 12,000 unique genes per cell. It is interesting to note that almost all of the cells have mitochondrial genes at 20 - 30% concentration compared to the rest of the genes. About 4 cells have even higher concentration of mitochondrial cells i.e. around 60% mitochondrial genes. Cells with a very high percentage of mitochondrial genes are considered as poor quality and we need to filter out these cells. 

```{r vp_mt, include=FALSE, echo=TRUE, warning=FALSE}
vp_mt <- VlnPlot(merged_seurat, features = c("nCount_RNA", "nFeature_RNA",          "percent.mt", ncol=3))
```

However, looking at each of these criteria individually can be misleading. The following scatter plot allows us to view both the feature count and the molecule count on the same graph. A good quality cell should follow a straight line as you would expect it to have a large number of molecules as well as large number of genes. From the graph below, we can see that most of the cells follow a straight line but some end up plateuing which could indicate not so good quality. We want to filter out such cells. 

```{r vp_filter, include=FALSE, echo=TRUE, warning=FALSE}

vp_filter <- FeatureScatter(merged_seurat, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + geom_smooth(method = lm)
```

*b) Calculate the transcript counts and filter for a set threshold count:*
```{r}
merged_subset <- subset(merged_seurat, subset = nCount_RNA > 800 & nFeature_RNA > 500 & percent.mt < 10)
```
After subsetting using some fixed threshold values, we see that the merged seurat object now contains 33538 feature across 67851 samples as opposed to 33538 genes across 77936. 





